src/rust/engine/fs/src/directory.rs

```diff
@@ -398,14 +434,31 @@ impl DigestTrie {
     Ok(Self(entries.into()))
   }
 
+  /// Create a DigestTrie from a root remexec::Directory and a map of its transitive children.
```

Comment:
Worth mentioning here that this is used for decoding `Tree`'s?

Reply:
Hm, yea. I think I could also just make it private, since it should only be used in `TryFrom<Tree>`. Will do in a followup.